scrypto/src/types/address.rs

```diff
@@ -63,6 +67,19 @@ impl Address {
         }
     }
 
+    // FIXME: This is a temporary interface. NFT Ids are u128, and need a
+    // simple way to map Address to the NFT Id space.
+    pub fn to_u128(&self) -> u128 {
+        let mut bytes: [u8; 16] = [0; 16];
+        match self {
+            Self::Package(d) => bytes[..].copy_from_slice(&d[..16]),
+            Self::Component(d) => bytes[..].copy_from_slice(&d[..16]),
+            Self::ResourceDef(d) => bytes[..].copy_from_slice(&d[..16]),
+            Self::PublicKey(d) => bytes[..].copy_from_slice(&d[..16]),
+        }
+        u128::from_be_bytes(bytes)
+    }
```

Comment:
I think we should move this to the `Account` blueprint before we finalise the NFT ID schema.

Reply:
Can I state the obvious... The overall system security for any transaction that would ever use an Account  is now degraded by truncating the public key by half.  This is almost certainly a bad idea that I wont elaborate on here in this comment, but it's an unnecessarily weak choice.  There are ways to make this translation cryptographically more secure if we absolutely must do comparisons of only 16 bytes everywhere else, but that also just seems like the "easy" choice and not a well designed choice.  These are security relevant decision points and should have some discussion.  -- Overall I like the design of this PR, but because it is extremely relevant it needs a more through review.

Reply:
Yes, we have certainly not missed this point. The fix to this will be allowing NFT ids be of greater length so that one may use 20 byte hash (ala bitcoin/eth) or the whole compressed public key for verification. This will be in a soon upcoming PR.

Reply:
Ok great.  Thanks for clarifying. I'll keep an eye out for the update.